topiary-cli/src/cli.rs

```diff
@@ -184,15 +184,20 @@ pub fn get_args() -> CLIResult<Cli> {
     let mut args = Cli::parse();
 
     // This is the earliest point that we can initialise the logger, from the --verbose flags,
-    // before any fallible operations have started
+    // before any fallible operations have started.
+    //
+    // Be more strict with logs coming with the library; prefer to display less of them.
+    let (global_filter_level, library_filter_level) = match args.global.verbose {
+        0 => (LevelFilter::Error, LevelFilter::Error),
+        1 => (LevelFilter::Warn, LevelFilter::Warn),
+        2 => (LevelFilter::Info, LevelFilter::Warn),
+        3 => (LevelFilter::Debug, LevelFilter::Warn),
+        _ => (LevelFilter::Trace, LevelFilter::Trace),
+    };
```

Comment:
I see the logic in hiding away the library logs, but I think this might fall under "surprising behaviour" for many. There are plenty of `info!` and `debug!` calls in the library; it would be a shame to obscure them unintentionally. I doubt there's much need for tuning the CLI and library logging levels independently -- adding something like `--lib-log-level` would definitely be too much, imo -- so my vote would be to keep these in parity.

Reply:
That makes sense to me.